nix/hetzner.nix

```diff
@@ -90,12 +90,177 @@ with lib;
+        parted --script -a optimal /dev/sdc -- mklabel gpt mkpart primary 1MiB 2MiB set 1 bios_grub on mkpart primary 2MiB '100%'
+        parted --script -a optimal /dev/sdd -- mklabel gpt mkpart primary 1MiB 2MiB set 1 bios_grub on mkpart primary 2MiB '100%'
+
+        # Now /dev/sd*1 is the BIOS boot partition, /dev/sd*2 is the one data partition
+
+        # Reload partition table so Linux can see the changes
+        partprobe
+
+        # Wait for all devices to exist
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sda1
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sda2
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdb1
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdb2
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdc1
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdc2
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdd1
+        udevadm settle --timeout=5 --exit-if-exists=/dev/sdd2
+
+        # --run makes mdadm not prompt the user for confirmation
+        mdadm --create --run --verbose /dev/md0 --level=1 --raid-devices=2 /dev/sda2 /dev/sdb2
+        mdadm --create --run --verbose /dev/md1 --level=1 --raid-devices=2 /dev/sdc2 /dev/sdd2
+
+        # Wipe filesystem signatures that might be on the RAID from some
+        # possibly existing older use of the disks.
+        # It's not clear to me *why* it is needed, but I have certainly
+        # observed that it is needed because ext4 labels magically survive
+        # mdadm RAID re-creations.
+        # See
+        #   https://serverfault.com/questions/911370/why-does-mdadm-zero-superblock-preserve-file-system-information
+        wipefs -a /dev/md0
+        wipefs -a /dev/md1
+
+        # Disable RAID recovery. We don't want this to slow down machine provisioning
+        # in the Hetzner rescue mode. It can run in normal operation after reboot.
+        echo 0 > /proc/sys/dev/raid/speed_limit_max
+
+        # `lazy_journal_init=1` to not have to zero the device;
+        # we use ATA TRIM with RZAT support to guarantee the device
+        # is already zeroed; see comment further up about the safety of that.
+        mke2fs -F -L rootjournal -O journal_dev -E lazy_journal_init=1 /dev/md0
+        mkfs.ext4 -F -L root -J device=/dev/md0 /dev/md1
+      '';
+      description = ''
+        Script to run after booting into the Hetzner rescue mode
+        to manually create partitions.
+
+        Note as of writing, Hetzner uses BIOS, not UEFI, so if you want
+        to use GPT partition tables (which you need in case you want to
+        make partitions larger than 2 TiB) you will likely have to make
+        a BIOS boot partition
+        (<link xlink:href="http://fedoraproject.org/wiki/Anaconda/Kickstart"/>).
+
+        Where possible, use the simpler "partitions" option instead of this option.
+
+        The "partitions" and "partitioningScript" options are mutually exclusive.
```

Comment:
`<option>partitions</option>` and `<option>partitioningScript</option>`

Reply:
Done